src/node_messaging.cc

```diff
@@ -186,27 +186,30 @@ uint32_t Message::AddWASMModule(WasmModuleObject::TransferrableModule&& mod) {
 
 namespace {
 
-void ThrowDataCloneException(Local<Context> context, Local<String> message) {
+MaybeLocal<Function> GetDOMException(Local<Context> context) {
   Isolate* isolate = context->GetIsolate();
-  Local<Value> argv[] = {
-    message,
-    FIXED_ONE_BYTE_STRING(isolate, "DataCloneError")
-  };
-  Local<Value> exception;
-
   Local<Object> per_context_bindings;
   Local<Value> domexception_ctor_val;
   if (!GetPerContextExports(context).ToLocal(&per_context_bindings) ||
       !per_context_bindings->Get(context,
                                 FIXED_ONE_BYTE_STRING(isolate, "DOMException"))
           .ToLocal(&domexception_ctor_val)) {
-    return;
+    return MaybeLocal<Function>();
   }
-
   CHECK(domexception_ctor_val->IsFunction());
   Local<Function> domexception_ctor = domexception_ctor_val.As<Function>();
-  if (!domexception_ctor->NewInstance(context, arraysize(argv), argv)
-          .ToLocal(&exception)) {
+  return domexception_ctor;
+}
+
+void ThrowDataCloneException(Local<Context> context, Local<String> message) {
```

Comment:
@addaleax I think ideally we'd want to do this in JS instead of throwing the error in nested C++ helpers?

Reply:
@joyeecheung We need to have this in C++, because the V8 ValueSerializer API requires it. I also don’t see any reason to move the error throwing to JS?

Reply:
I see, this is required by `ValueSerializer::Delegate::ThrowDataCloneError` - but do we need to throw errors this way ourselves? As this is just doing `new DOMException(message, 'DataCloneError')` - I believe in general we prefer [throwing errors from JS](https://github.com/nodejs/node/blob/master/CPP_STYLE_GUIDE.md#avoid-throwing-javascript-errors-in-nested-c-methods).